android-app-examples/SNetDemo/app/src/main/java/com/example/snetdemo/ImageSegmentationActivity.java

```diff
@@ -0,0 +1,449 @@
+package com.example.snetdemo;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.ContextCompat;
+import androidx.core.content.FileProvider;
+
+import android.Manifest;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.pm.ActivityInfo;
+import android.content.pm.PackageManager;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+import com.google.protobuf.ByteString;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigInteger;
+import java.util.Calendar;
+
+import io.singularitynet.service.semanticsegmentation.Segmentation;
+
+import static com.example.snetdemo.ImageUtils.*;
+
+
+public class ImageSegmentationActivity extends AppCompatActivity
+{
+
+    final int REQUEST_CODE_UPLOAD_INPUT_IMAGE = 10;
+    final int REQUEST_CODE_IMAGE_CAPTURE = 11;
+
+    final int PROGRESS_WAITING_FOR_SERIVCE_RESPONSE = 2;
+    final int PROGRESS_DECODING_SERIVCE_RESPONSE = 3;
+    final int PROGRESS_LOADING_IMAGE = 4;
+    final int PROGRESS_FINISHED = 5;
+
+    private boolean mIsInputImageUploaded = false;
```

Comment:
Minor, but in Java member names are not prefixed usually as there is no global variables at all.